hw4.Rmd

```diff
@@ -30,43 +30,70 @@ Before doing anything else, just play with the data to get a sense of it.
 (@) The main outcome of interest is *decision* which records whether the judge granted an appeal or not.  In what proprotion of cases did the judge grant an appeal.
 
 ```{r}
-# [your code here]
+counts <- table(data$decision)
+prop.table(counts)
 ```
 
-(@) There are 12 different judges in the data. A key question is whether different judges have different rates of granting an appeal.  Make a plot showing the number of cases per judge.
+(@) There are 10 different judges in the data. A key question is whether different judges have different rates of granting an appeal.  Make a plot showing the number of cases per judge.
 
 ```{r}
-# [your code here]
+library(ggplot2)
+
+h <- ggplot(data, aes(x = judge))
+h + geom_bar()
 ```
 
 (@) Now plot the rate at which the judges grant appeals. 
 
 ```{r}
-# [your code here]
+library(dplyr)
+data$decision <- ifelse(data$decision == "yes", 1, 0)
+
+data <- data %>%
+  group_by(judge) %>%
+  mutate(rate.appeal = mean(decision))
+
+k <- ggplot(data, aes(x = judge, y = rate.appeal))
+k + geom_point()
 ```
 
 (@) Now let's try this with logistic regression because we are going to move to more complicated models.  Fit a logistic regression predicting whetheran appeal will be granted using judge as a categorical predictor. Use Iacobucci as the reference judge to facilitate the comparisons that follow. For more on how to control the reference level of a factor, check out this post on StackOverflow: http://stackoverflow.com/questions/3872070/how-to-force-r-to-use-a-specified-factor-level-as-reference-in-a-regression
 
 ```{r}
-# [your code here]
+data$judge <- relevel(data$judge, ref = "Iacobucci" )
+
+fit <- glm(decision ~ judge, data = data, family ="binomial")
+
+summary(fit)
 ```
 
 (@) Using the results of the model calculate the predicted probability that Judge Marceau will grant an appeal.
 
 ```{r}
-# [your code here]
+coefs <- coef(fit)
+prob.a <- 1/(1 + exp(-(coefs[1] + coefs[6]*1)))
```

Comment:
coefs[6] calls the coefficient for Mahoney, not Marceau. This is a good example of why you want to avoid calling values with position numbers which can change. See solutions for a robust alternative.